### PR TITLE
feat(gatsby-source-shopify): add metafields to product nodes

### PR DESCRIPTION
**Pull request by kellyvaughn**
## Description

Shopify introduced access to metafields via the Storefront API back in May. This adds the metafield endpoint to products via gatsby-source-shopify.

## Related Issues

N/A

### CODE REVIEW

File: packages/gatsby-source-shopify/src/nodes.js [hunk 1]
```diff
@@ -129,6 +138,9 @@ export const ProductNode = imageArgs =>
     return node
   })
 
+export const ProductMetafieldNode = _imageArgs =>
```
Comment (DSchau) [inline packages/gatsby-source-shopify/src/nodes.js]: Not really new here -- but I wonder why this pattern is used through all of these node types?

Oh well!

Comment (kellyvaughn) [reply packages/gatsby-source-shopify/src/nodes.js]: That's a really good question - I didn't want to reinvent anything, so I just followed the same pattern as before. I don't totally understand it to be entirely honest.

Comment (DSchau) [reply packages/gatsby-source-shopify/src/nodes.js]: 🤷‍♂ Your guess is as good as mine. Not hurting anything -- we'll clean it up later!

**Comment by DSchau** [comment]
Published as [`gatsby-source-shopify`@`2.2.0`](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-source-shopify/CHANGELOG.md#220-2019-08-01).

Enjoy the swag 🎉

**Pull request merged**